sdk-core/src/main/java/care/data4life/sdk/network/NetworkingContract.kt

```diff
@@ -117,28 +140,63 @@ interface NetworkingContract {
         fun fromName(name: String?): Environment
     }
 
+    internal interface ClientFactory {
+        fun getInstance(
+            authService: AuthorizationContract.Service,
+            environment: Environment,
+            clientId: String,
+            clientSecret: String,
+            platform: String,
+            connectivityService: NetworkConnectivityService,
+            clientName: Clients,
+            clientVersion: String,
+            staticAccessToken: ByteArray?,
+            debugFlag: Boolean
+        ): OkHttpClient
+    }
+
+    internal interface IHCServiceFactory {
+        fun getInstance(
+            client: OkHttpClient,
+            platform: String,
+            environment: Environment
+        ): IHCService
+    }
+
+    interface SearchTagsBuilder {
+        fun addOrTuple(tuple: List<String>): SearchTagsBuilder
+        fun seal(): SearchTags
+    }
+
+    interface SearchTags {
+        val tags: String
+    }
+
+    interface SearchTagsBuilderFactory {
+        fun newBuilder(): SearchTagsBuilder
+    }
+
     companion object {
         const val PLATFORM_D4L = "d4l"
         const val PLATFORM_S4H = "s4h"
         const val DATA4LIFE_CARE = "sha256/AJvjswWs1n4m1KDmFNnTqBit2RHFvXsrVU3Uhxcoe4Y="
         const val HPSGC_DE = "sha256/3f81qEv2rjHvcrwof2egbKo5MjjSHaN/4DOl7R+pH0E="
         const val SMART4HEALTH_EU = "sha256/yPBKbgJMVnMeovGKbAtuz65sfy/gpDu0WTiuB8bE5G0="
-        const val REQUEST_TIMEOUT = 2
+        const val REQUEST_TIMEOUT: Long = 2
         const val HEADER_ALIAS = "gc_alias"
-        const val HEADER_ACCESS_TOKEN = "access_token"
         const val HEADER_AUTHORIZATION = "Authorization"
-        const val HEADER_BASIC_AUTH = "basic_auth"
-        const val HEADER_GC_SDK_VERSION = "GC-SDK-Version"
+        const val ACCESS_TOKEN_MARKER = "access_token"
+        const val BASIC_AUTH_MARKER = "basic_auth"
+        const val HEADER_GC_SDK_VERSION = "gc-sdk-version"
```

Comment:
please remember to change this (i guess its done in the other PR)

Reply:
it is